dao/application_authentication_dao.go

```diff
@@ -11,6 +11,46 @@ type ApplicationAuthenticationDaoImpl struct {
 	TenantID *int64
 }
 
+func (a *ApplicationAuthenticationDaoImpl) ApplicationAuthenticationsByApplications(applications []m.Application) ([]m.ApplicationAuthentication, error) {
```

Comment:
What do you think about renaming this to `ByApplications`? Sine we're in the `ApplicationAuthenticationDaoImpl` I guess we can assume that the query is to fetch appauths. Same about the other two functions, which I guess they could be renamed to `ByAuthentications` and `ByResource` accordingly.

Reply:
The word `map`/`get` comes to mind too. `GetFromApplications` and/or `MapFromApplications`.

Reply:
naming is hard and in this case it looks that origin suggestion seems to me still more clear (I am ok with changing `By` -> `From` as it is probably correct from English language )

so we have signature:

```go
func (a *ApplicationAuthenticationDaoImpl) ApplicationAuthenticationsByApplications(applications []m.Application)
```

** Removing `ApplicationAuthentications` because type of receiver is "same"**
but receiver is not same type it just contains `ApplicationAuthentication` and even it is in singular form - so it is different type and there is also confusion about plurals. 

**Include Get or Map in name of method**

**Get**
I don't prefer prefix `Get` because this word in here doesn't say anything in addition there is no added value (example):  

```
aa := a.ApplicationAuthentications(..)
```
vs

```
aa := a.GetApplicationAuthentications(..)
```

it is obvious that in both cases returned values are` ApplicationAuthentications`

**Map**
 
Map is good word in method naming but here 
It feels that `Map` doesn't cover exactly the meaning - maps are types in GO (end even there are not used it this method ) and 
 in this method is mapping is not happening there is just query


** By -> From *
I am ok with this.

NOTE: personally I would remove also `...Applications` because it is stated in first parameter but I agreed with Jacob that it should stay here because it is more obvious what is method about.

Naming is hard and important - so thanks for your inputs - it is important how other perceives it - we definitely need all to agree on this - so we can use this pattern across whole application.

let me know what do you think about it now after my comments, thanks!

Reply:
> **Removing `ApplicationAuthentications` because type of receiver is "same"**
> but receiver is not same type it just contains `ApplicationAuthentication` and even it is in singular form - so it is different type and there is also confusion about plurals.

I was referring to the fact that the function's signature already gives a clue about what we will be returning: `([]m.ApplicationAuthentication, error)`. So by taking a look at the signature we know that an array of "ApplicationAuthentication"s will be returned. And if we comment the function then it'd be even more clear.


It is true though that the struct could be named anything, like `type DifferentNameDaoImpl struct` and make it implement the "ApplicationAuthenticationDao" interface, which would make using the `List()`, `Create()`, `GetBy` functions unclear: what would we be listing? DifferentNames?

However this is an hypothetical case, which I guess we wouldn't allow in a code review.

---

Regarding standardizing a convention for naming methods, I do really like how Spring names the operations in their CRUD repositories\[1\]:

```java
    <S extends T> S save(S entity);
    T findOne(ID primaryKey);
    Iterable<T> findAll();
    Long count();
    void delete(T entity);
    boolean exists(ID primaryKey);
```

Plus the signatures for functions like the one we're discussing about would be like `FindAllByApplications(...)`, which maks it clear that it is going to find all the AppAuths by the applications you provide.

Though I'm feeling that this PR shouldn't be the place to discuss this.

\[1\]: https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html

---

So, my final vote is: let's leave it like it is, since discussing the naming of the functions isn't the goal of this PR. I'm sorry for diverting the discussion.

Reply:
thanks @MikelAlejoBR for your insights , I understand now and I like `FindAll` pattern but let's leave it for follow up PRs.